src/kaleidoscope/plugin/EEPROM-Settings.cpp

```diff
@@ -54,21 +54,30 @@ uint8_t EEPROMSettings::default_layer(uint8_t layer) {
     Layer.move(layer);
     settings_.default_layer = layer;
   }
-  if (layer == 0xff) {
-    settings_.default_layer = layer;
+  if (layer >= 0x7e) {
+    settings_.default_layer = 0x7e;
   }
   update();
   return settings_.default_layer;
 }
 
+void EEPROMSettings::use_eeprom_layers_only(bool value) {
+  settings_.use_eeprom_layers_only = value;
+  if (settings_.default_layer >= 0x7e)
+    settings_.default_layer = 0x7e;
+  update();
+}
+
 void EEPROMSettings::seal(void) {
   sealed_ = true;
 
   CRC.finalize();
 
-  /* If we have a default layer set, switch to it. As 0xff is the default EEPROM
-   * value, treat it as not having a default layer set. */
-  if (settings_.default_layer != 0xff)
+  /* If we have a default layer set, switch to it. 0x7f is the default EEPROM
+   * value (since the highest bit is not ours), 0x7e is the explicit "don't mess
+   * with the default layer" setting. In either of these cases, we skip moving
+   * to the default layer. */
+  if (settings_.default_layer < 0x7e)
```

Comment:
Oh hey. you do explain it :) But extract the constant and put the docs on the constant :)